app/api/v2/managers/operation_api_manager.py

```diff
@@ -1,16 +1,85 @@
+import uuid
+
 from app.api.v2.managers.base_api_manager import BaseApiManager
-from app.api.v2.responses import JsonHttpNotFound, JsonHttpForbidden
+from app.api.v2.responses import JsonHttpNotFound, JsonHttpForbidden, JsonHttpBadRequest
+from app.objects.secondclass.c_link import Link
+from app.objects.secondclass.c_executor import Executor
+from app.objects.c_ability import Ability
+from app.objects.c_agent import Agent
+from app.objects.c_operation import Operation
+from app.utility.base_world import BaseWorld
 
 
 class OperationApiManager(BaseApiManager):
-    def __init__(self, data_svc, file_svc):
-        super().__init__(data_svc=data_svc, file_svc=file_svc)
+    def __init__(self, services):
+        super().__init__(data_svc=services['data_svc'], file_svc=services['file_svc'])
+        self.services = services
 
     async def get_operation_report(self, operation_id: str, access: dict):
         operation = await self.get_operation_object(operation_id, access)
         report = await operation.report(file_svc=self._file_svc, data_svc=self._data_svc)
         return report
 
+    async def get_operation_links(self, operation_id: str, access: dict):
+        operation = await self.get_operation_object(operation_id, access)
+        links = [link.display for link in operation.chain]
+        return links
+
+    async def get_operation_link(self, operation_id: str, link_id: str, access: dict):
+        operation = await self.get_operation_object(operation_id, access)
+        link = self.search_operation_for_link(operation, link_id)
+        return link.display
+
+    async def update_operation_link(self, operation_id: str, link_id: str, link_data: dict, access: BaseWorld.Access):
+        operation = await self.get_operation_object(operation_id, access)
+        link = self.search_operation_for_link(operation, link_id)
+        if link.access not in access['access']:
+            raise JsonHttpForbidden(f'Cannot update link {link_id} due to insufficient permissions.')
+        if link.is_finished():
+            raise JsonHttpForbidden(f'Cannot update a finished link: {link_id}')
```

Comment:
Right now I have discarded links as being 'finished' in that method, but what does the 'high-viz' status mean?

Reply:
I believe high-viz means the link wasn't run because the visibility was too high (it's like our op-sec setting. if we want our operation to run only stealthy links, it'll skip links with too high of a visibility value)

Reply:
Thanks! It's been updated to check for high-viz/"ignorable" links